src/ipz_parser.cpp

```diff
@@ -441,4 +441,148 @@ types::VPDMapVariant IpzVpdParser::parse()
     }
 }
 
+bool IpzVpdParser::findRecordInVTOC(
+    types::BinaryVector::const_iterator& io_itrToVPD, const auto i_ptLength,
+    const auto i_record)
+{
+    auto l_itrToVTOCEnd = io_itrToVPD;
+    std::ranges::advance(l_itrToVTOCEnd, i_ptLength, m_vpdVector.cend());
+
+    // Go through VTOC to find the given record's offset.
+    while (io_itrToVPD < l_itrToVTOCEnd)
+    {
+        std::string l_recordName(io_itrToVPD,
+                                 io_itrToVPD + Length::RECORD_NAME);
+
+        // Continue to next record until we find the record which we are
+        // interested
+        if (l_recordName != i_record)
+        {
+            std::ranges::advance(
+                io_itrToVPD,
+                Length::RECORD_NAME + sizeof(types::RecordType) +
+                    sizeof(types::RecordOffset) + sizeof(types::RecordLength) +
+                    sizeof(types::ECCOffset) + sizeof(types::ECCLength),
+                m_vpdVector.cend());
+            continue;
+        }
+
+        // Skip record name and record type
+        std::ranges::advance(io_itrToVPD,
+                             Length::RECORD_NAME + sizeof(types::RecordType),
+                             m_vpdVector.cend());
+
+        // Record found
+        return true;
+    }
+    return false;
+}
+
+types::BinaryVector
+    IpzVpdParser::getKeywordValueFromHW(const types::Record& i_record,
```

Comment:
it does get keyword's value but from a vpd vector  not hardware

Reply:
better to be consistent with your above function , you could have named it getKeywordValueFromRecord